### DISCUSSION TITLE
[Feature] iOS shortcuts support

### DISCUSSION THREAD

**Discussion by iswaterbad** [immich-app/immich#11297] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Having support for apple shortcuts would let users set their own rules for when to back up photos in the background. This allows for more customization on the conditions of when to back up photos.

The [photosync](https://apps.apple.com/us/app/photosync-transfer-photos/id415850124) app has a similar feature (it’s a paid app though). 

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by radh21301** [comment]
This would be awesome to have. Maybe it can help with the horrible background backups on IOS!

**Comment by Encotric** [comment]
Hi, I started working on a [PR](https://github.com/immich-app/immich/pull/12915) for this feature. I hope it will work using the current background task solution, otherwise it would require a lot of new work, unfortunately.

**Comment by exelix11** [comment]
For android feature parity i'd also like to request integration with Tasker, Samsung's modes and routines and similar automation apps. On android this can be accomplished with a static [app shortcut](https://developer.android.com/develop/ui/views/launch/shortcuts) combined with an exported intent.

This would allow to configure when to run backups with extreme flexibility instead of relying on the current foreground/background logic.

**Comment by EthanG45** [comment]
This service is a non-starter for iOS without this. I'd love to see a backup and delete system where we can run a shortcut to back up our photos and pair it with a shortcut to delete the photos from Apple Photos (this already exists).

**Comment by bo0tzz** [comment]
I've edited the title to make this a request for general shortcuts support

**Comment by elevatebart** [comment]
I have a working implementation of this and would like to know whether it's something you'd accept before I polish it into a PR.

Scope is four App Intents, all Swift, no Dart changes:

Get Random Photo — album / orientation / quality parameters, returns a file. Chains into the system Set Wallpaper action, which covers #10097 and #14299.
Get Memory Photo — a photo from today's memories.
Search Photos — smart search by description, returns a list.
Back Up My Library — runs the same worker as the scheduled BGProcessingTask, so uploads go through startBackupWithURLSession and continue with the app suspended. The point is deterministic timing: a nightly automation at a chosen hour instead of waiting on BGTaskScheduler.

The first three live in the existing WidgetExtension and reuse its ImmichAPI client and Album entity picker, so they cost almost nothing. The backup one adds a runOnDemand entry point next to the existing BGTask handlers.

Two things I want to flag rather than have you discover in review.

Android parity. You asked about this on #12915 and I don't have an answer. Nothing here helps Android, and I'm not set up to build the App Actions / Tasker side. If parity is a precondition for merging, that's worth saying now, before anyone spends review time on this.

I could not fully test it. 
I don't have a paid apple dev account.
It built and tests passed.

Happy to go either way, or to split it into separate PRs. Mostly I'd like to know if there's appetite before going further, given #12915 went stale.

**Comment by elevatebart** [comment]
Branch is up if it's easier to judge from the diff: https://github.com/immich-app/immich/compare/main...elevatebart:immich:elevate-bot/immich-ios-shortcuts-wallpaper-b402c4

I only changed the Swift code, that's apple specific for now. The three read-only intents come in addition to the existing `WidgetExtension` and use the same API client; the backup one adds a `runOnDemand` entry point next to the current BGTask handlers. Nothing in the backup system is refactored — I split that out after reading your note on #12915.
